pacsman/pynetdicom_client.py

```diff
@@ -81,29 +84,10 @@ def search_patients(self, search_query, additional_tags=None):
                     #  (some dupes are returned, especially for ID search)
                     uid_to_result[dataset.StudyInstanceUID] = dataset
 
-            # separate by patient ID, count studies and get most recent
-            patient_id_to_datasets = {}
+            patient_id_to_datasets = defaultdict(Dataset)
             for study in uid_to_result.values():
-                patient_id = study.PatientID
```

Comment:
Similarly to how I did here, I think we may be able to pull out logic from the file system and pynet dicom backends and into the base client.

Reply:
Perhaps next time we touch the other methods we can look into doing this.